### DISCUSSION TITLE
How do project Pages work?

### DISCUSSION THREAD

**Discussion by blueglyph** [skills/.github#208] [Q&A] [answered]
I'm new to GitHub pages and I'm trying to understand how this works for project pages (**NOT** the personal Pages site "{username}.github.io" repository).

I tried the [tutorial](https://github.com/skills/github-pages) but it wasn't working, and it uses a template workflow. Besides, it doesn't explain much. I saw that it created a new `my-pages` branch where it created the empty `_config.yml`, and a basic `index.md` - which, contrary to the explanation, is not visible when looking at the published site, only the `README.md` is (with a very bad formatting and missing images).

If I activate Pages on an existing project, without adding any workflow, I see that the README.md is automatically translated into a static website. I think that it's done by a version of Jekyll used by an automatic flow on GitHub. The site is visible and fine, but what is generated is inaccessible - no branch or file created, nothing is visible from the Actions pages, and I didn't find any doc on this. Everything is focused on the personal pages.

- Can we rely on this project automatic workflow, and is it possible to configure the style and so on?
- Or are we supposed to always use Jekyll to generate the content?
- Does Jekyll work with project pages too? Is there some doc I missed about it?

**Comment by sinsukehlab** [comment]
The course works quite well. Let me explain.
Published sites are **NOT** Git repositories.
The index page is not visible at first because the GitHub Pages site is deployed from the `main` branch. The `main` branch doesn't contain `index.md` at first, so the published site displays `README.md`.
You are taught to edit the files in the `my-pages` branch, create and merge a PR from `my-pages` to `main`. When the commits pushed to the `main` branch, the GitHub Actions workflow 'pages build and deployment' runs to publish the site.

**Reply by blueglyph** [reply]
Thanks!

I see. But the merge is the final step, so why does it do all those build and deployment at every step in the Actions tab if there is nothing to deploy yet? The tutorial instructs to open the website, but I don't see the point if there's nothing worth seeing yet except a broken page.

To me, all that and the fact the website changes at each step suggests that the workflow publishes each modification and that `my_pages` is the link between the user's input in `main` and how the site is rendered.

The empty `_config.yml` file is suspicious too; once again, it seems in contradiction with what is mentioned elsewhere.

This is confusing at best, and doesn't explain anything much. It doesn't tell whether Jekyll can be used for project pages either, nor how. The first step to improve this tutorial, I think, would be to remove all these mysterious workflows to avoid unnecessary noise and confusion, and to present a realistic environment to the user.

Anyway, the tutorial was only anecdotic to my question, but now I know that `my_pages` is just an artefact of those custom tutorial scripts.

**Reply by sinsukehlab** [reply]
What do you mean by 'project pages'? Every repository in GitHub can have a GitHub Pages site deployed from a branch if set properly.
The workflows are not mysterious. They just update steps in `README.md`.
Any changes to the content are usually made on a newly created branch. You merge the branch to the `main` branch when it is ready.
You are expected to merge the `my-pages` branch to the `main` branch after creating a blog post in this course.

**Reply by blueglyph** [reply]
There are 3 types of GitHub pages: user, project and organization.

You're missing my point. Thanks for the help, but let's see if someone else has some answers.

**Reply by sinsukehlab** [reply]
You get one site per GitHub account and organization, and unlimited project sites using GitHub Pages.
This course will publish a project site unless you generate a repository named 'blueglyph.github.io' from the template repository.
You have published a project site.
See: https://pages.github.com/

**Reply by blueglyph** [reply]
Please stop trying to help, because you're not answering my questions. I don't care about the damn tutorial!

**Comment by blueglyph** [comment, marked as answer]
Anyway, I've learned that Jekyll was apparently automatically used with GitHub pages and that it was a pretty old version, so I'll use another static site generator and handle that myself. It remove all that mystery part from the equation.
